src/Interpreters/InterpreterShowTablesQuery.cpp

```diff
@@ -214,7 +214,7 @@ BlockIO InterpreterShowTablesQuery::execute()
         return res;
     }
 
-    return executeQuery(getRewrittenQuery(), getContext(), true).second;
+    return executeQuery(getRewrittenQuery(), getContext(), QueryFlags(true, false)).second;
```

Comment:
Also maybe we can provide default values for the flags, and set only changed ones like `QueryFlags{ .internal = true }`